zebra-consensus/src/block/check.rs

```diff
@@ -169,9 +169,25 @@ pub fn time_is_valid_at(
 ///
 /// `transaction_hashes` is a precomputed list of transaction hashes.
 pub fn merkle_root_validity(
```

Comment:
Remember to update the function docs when you add a new consensus rule:
* quote the consensus branch id rule and link to its spec section: `nConsensusBranchId must match...` in https://zips.z.cash/protocol/nu5.pdf#txnencodingandconsensus
* since the merkle tree rule is missing, also quote that rule, and link to its spec section: `hashMerkleRoot` in https://zips.z.cash/protocol/nu5.pdf#blockheader

Since you've added a new panic (`expect`) to the function, add a `Panics` section to the function docs:
* if `block` does not have a coinbase transaction

Reply:
I was not able to understand the merke tree rule comment we need to do. I Left a spot for it in https://github.com/ZcashFoundation/zebra/pull/2100/commits/6a5dd7f5f444ae7c3697bbadb59d526a74f2a8c5